### PR TITLE
Clean up some rawls compile warnings

### PR DESCRIPTION
**Pull request by davidangb**
no ticket.

Code cleanup - primarily migration from `JdbcDriver` to `JdbcProfile` - to get rid of *some* of the compile warnings. This doesn't come close to getting rid of all of them nor does this PR intend to do so ... incremental progress!

I thought about tackling the remainder of the `JavaConverters` deprecation warnings, but that would have made this PR much larger; I'd rather make a separate PR for that.

Read individual commits if desired to review smaller chunks.

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala [hunk 1]
```diff
@@ -67,7 +67,7 @@ case class HttpClientUtilsInstrumented(implicit val materializer: Materializer,
   }
 }
 
-case class HttpClientUtilsGzipInstrumented(implicit val materializer: Materializer, requestCounter: (HttpRequest, HttpResponse) => Counter, requestTimer: (HttpRequest, HttpResponse) => Timer, actorRefFactory: ActorRefFactory, val executionContext: ExecutionContext) extends HttpClientUtils {
+case class HttpClientUtilsGzipInstrumented()(implicit val materializer: Materializer, requestCounter: (HttpRequest, HttpResponse) => Counter, requestTimer: (HttpRequest, HttpResponse) => Timer, actorRefFactory: ActorRefFactory, val executionContext: ExecutionContext) extends HttpClientUtils {
```
Comment (ghost) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala:70]: Minor nitpick: can remove these empty parens (I think this was IntelliJ)

**Review by ghost — APPROVED** [review]
Compiled with less warnings!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala (hunk 1 above)
Comment (davidangb) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala:70]: I added these explicitly to remove a compile warning! Something along the lines of "case classes should have an explicit argument list"

Comment (ghost) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala:70]: Oh good to know!

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala [hunk 2]
```diff
@@ -368,11 +368,11 @@ trait WorkspaceComponent {
     }
 
     def getUserSharePermissions(subjectId: RawlsUserSubjectId, workspaceContext: SlickWorkspaceContext): ReadWriteAction[Boolean] = {
-      workspaceUserShareQuery.filter(rec => rec.userSubjectId === subjectId.value && rec.workspaceId === workspaceContext.workspaceId).countDistinct.result.map(rows => rows > 0) flatMap { hasSharePermission =>
```
Comment (jmthibault79) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala]: Here we need to be careful we're not relying on the older behavior: https://github.com/slick/slick/issues/1651

File: core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala (hunk 1 above)
Comment (jmthibault79) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala:70]: Can they be `case object`s then?  Not sure if that's compatible with the implicits.

Comment (davidangb) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/util/HttpClientUtils.scala:70]: I'm pretty certain that would not work with the implicits.

**Comment by davidangb** [comment]
@jmthibault79 I've reverted the `countDistinct` change and will refactor that separately, that work for you?

**Review by jmthibault79 — APPROVED** [review]

**Comment by davidangb** [comment]
`rawls-build` job was mysteriously hosed and I aborted it after 3hr 40min. Let's try again: jenkins retest.

**Pull request merged**